services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/accessSurrogates/AccessSurrogateRequestProcessor.java

```diff
@@ -0,0 +1,80 @@
+package edu.unc.lib.boxc.services.camel.accessSurrogates;
+
+import edu.unc.lib.boxc.auth.api.Permission;
+import edu.unc.lib.boxc.auth.api.services.AccessControlService;
+import edu.unc.lib.boxc.model.api.DatastreamType;
+import edu.unc.lib.boxc.model.api.objects.FileObject;
+import edu.unc.lib.boxc.model.api.objects.RepositoryObjectLoader;
+import edu.unc.lib.boxc.model.fcrepo.ids.PIDs;
+import edu.unc.lib.boxc.model.fcrepo.services.DerivativeService;
+import edu.unc.lib.boxc.operations.jms.MessageSender;
+import edu.unc.lib.boxc.operations.jms.accessSurrogates.AccessSurrogateRequestSerializationHelper;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.jdom2.Document;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Objects;
+
+import static edu.unc.lib.boxc.operations.jms.RunEnhancementsMessageHelpers.makeEnhancementOperationBody;
+import static edu.unc.lib.boxc.operations.jms.accessSurrogates.AccessSurrogateRequest.DELETE;
+import static edu.unc.lib.boxc.operations.jms.accessSurrogates.AccessSurrogateRequest.SET;
+import static org.apache.commons.io.file.PathUtils.deleteFile;
+
+/**
+ * Processor for requests for adding, replacing, and removing access surrogates
+ *
+ * @author snluong
+ */
+public class AccessSurrogateRequestProcessor implements Processor {
+    private AccessControlService accessControlService;
+    private RepositoryObjectLoader repositoryObjectLoader;
+    private DerivativeService derivativeService;
+    private MessageSender messageSender;
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        var in = exchange.getIn();
+        var request = AccessSurrogateRequestSerializationHelper.toRequest(in.getBody(String.class));
+        var agent = request.getAgent();
+        var pid = PIDs.get(request.getPidString());
+
+        accessControlService.assertHasAccess("User does not have permission to update access surrogates",
+                pid, agent.getPrincipals(), Permission.editDescription);
+
+        var action = request.getAction();
+        var repositoryObject = repositoryObjectLoader.getRepositoryObject(pid);
+        if (repositoryObject instanceof FileObject) {
```

Comment:
I wonder if we should log a error when something that isn't a FileObject is received? Feels a little weird to just ignore the request.

Reply:
ah yeah, i was wondering what to do about that. don't know why i didn't think of just logging it. would it be an error or a warn?